lms/models/h_user.py

```diff
@@ -28,3 +29,22 @@ class HUser(NamedTuple):
 
     def userid(self, authority):
         return f"acct:{self.username}@{authority}"
+
+    @classmethod
+    def from_lti_user(cls, lti_user):
+        provider = lti_user.tool_consumer_instance_guid
+        provider_unique_id = lti_user.user_id
+
+        def username():
+            """Return the h username for the current request."""
+            username_hash_object = hashlib.sha1()
+            username_hash_object.update(provider.encode())
+            username_hash_object.update(provider_unique_id.encode())
+            return username_hash_object.hexdigest()[:30]
+
+        return cls(
+            username=username(),
+            display_name=lti_user.display_name,
+            provider=provider,
+            provider_unique_id=provider_unique_id,
+        )
```

Comment:
The method for constructing `HUser`'s belongs on the `HUser` class, but I've also added `LTIUser.h_user` (below) as a convenience

Reply:
Getting some deja-vu here, but I think the inner method should just be a plain method like `lti_user.py/display_name()`.

Also, once this is true, this factory method (and the attendant boundary and tests) doesn't need to exist. The `display_name()` part probably belongs here, but the rest can be in the `h_user()` property on the `lti_user`, as it would just be a call to the initialiser, with a call to this function first.